lib/bson/decimal128.js

```diff
@@ -290,21 +307,14 @@ Decimal128.fromString = function(string) {
       }
     }
 
-    if (foundNonZero) {
-      nDigits = nDigits + 1;
-    }
-
-    if (sawRadix) {
-      radixPosition = radixPosition + 1;
-    }
+    if (foundNonZero) nDigits = nDigits + 1;
+    if (sawRadix) radixPosition = radixPosition + 1;
 
     nDigitsRead = nDigitsRead + 1;
     index = index + 1;
   }
 
-  if (sawRadix && !nDigitsRead) {
-    throw new Error('' + string + ' not a valid Decimal128 string');
-  }
+  if (sawRadix && !nDigitsRead) throw new Error('' + string + ' not a valid Decimal128 string');
```

Comment:
perhaps we should wrap the string in quotes?
```
throw new Error(`"${string}" not a valid Decimal128 string`);
```

Reply:
Also note I used template strings here, it's a convenient way to not need that imported `f` format function above.